packages/theme-product-form/__tests__/theme-product-form.test.js

```diff
@@ -0,0 +1,311 @@
+      )
+    ).toBe('https://shop1.myshopify.com?id=12345678&variant=87654321');
+  });
+});
+
+describe('ProductForm()', () => {
+  test('is a constructor', () => {
+    const element = document.getElementById('form');
+    const productForm = new ProductForm(element, productJSON);
+
+    expect(productForm).toBeInstanceOf(ProductForm);
+  });
+
+  test('throws an error if the first argument is not a DOM element', () => {
+    expect(() => new ProductForm(null, productJSON)).toThrowError(TypeError);
+  });
+
+  test('throws an error if the second argument is not a valid product object', () => {
+    const element = document.getElementById('form');
+    expect(() => new ProductForm(element)).toThrowError(TypeError);
+  });
+
+  test('assigns the form element that was passed as the first argument to .element', () => {
+    const element = document.getElementById('form');
+    const productForm = new ProductForm(element, productJSON);
+
+    expect(productForm.element).toBe(element);
+  });
+
+  test('assigns the product object that is passed as the seconds argument to .product', () => {
+    const element = document.getElementById('form');
+    const productForm = new ProductForm(element, productJSON);
+
+    expect(productForm.element).toBe(element);
+  });
+
+  test('assigns all option inputs which have a name attribute that start with "options" to .optionInputs', () => {
+    const element = document.getElementById('form');
+    const productForm = new ProductForm(element, productJSON);
+
+    expect(productForm.optionInputs.length).toBe(4);
+  });
+
+  test('assign all quantity inputs which have the name attribute equal to "quantity" to .quantityInputs', () => {
+    const element = document.getElementById('form');
+    const productForm = new ProductForm(element, productJSON);
+
+    expect(productForm.quantityInputs.length).toBe(1);
+  });
+
+  test('assign all property inputs which have the name attribute that start with "properties" to .propertyInputs', () => {
+    const element = document.getElementById('form');
+    const productForm = new ProductForm(element, productJSON);
+
+    expect(productForm.propertyInputs.length).toBe(defaultProperties.length);
+  });
+
+  test('calls the method assigned to onOptionChange option when the value of an option input changes', () => {
+    const element = document.getElementById('form');
+    const colorSelect = document.querySelector('[name="options[Color]"]');
+    const sizeSelect = document.querySelector('[name="options[Size]"]');
+    const changeEvent = new Event('change');
+    const config = {
+      onOptionChange: jest.fn()
+    };
+    const options = [
+      {name: 'Color', value: 'Cobalt'},
+      {name: 'Voltage', value: '220 Volts'},
+      {name: 'Size', value: 'Large'}
+    ];
+
+    // eslint-disable-next-line no-unused-vars
+    const productForm = new ProductForm(element, productJSON, config);
+
+    sizeSelect.value = 'Large';
+    colorSelect.value = 'Cobalt';
```

Comment:
I see that in the `onChangeProperties` test below, you opted to use the `properties` variable to assign your new value, is there a reason you're hard coding it here?
e.g., `propertyElement.value = properties[0].value;`